src/model/HeatPumpPlantLoopEIRCooling.cpp

```diff
@@ -0,0 +1,626 @@
+    }
+
+    bool HeatPumpPlantLoopEIRCooling_Impl::addToNode(Node& node) {
+
+      // call the base class implementation to connect the component
+      bool ok = WaterToWaterComponent_Impl::addToNode(node);
+
+      // If there's a secondary plant loop, switch the condenser type to "WaterCooled"
+      if (this->sourceSideWaterLoop()) {
+        this->setCondenserType("WaterSource");
+      }
+      return ok;
+    }
+
+    bool HeatPumpPlantLoopEIRCooling_Impl::removeFromSecondaryPlantLoop() {
+      // Disconnect the component
+      bool ok = WaterToWaterComponent_Impl::removeFromSecondaryPlantLoop();
+
+      // Switch the condenser type back to "AirSource"
+      this->setCondenserType("AirSource");
+      return ok;
+    }
+
+    std::string HeatPumpPlantLoopEIRCooling_Impl::condenserType() const {
+      boost::optional<std::string> value = getString(OS_HeatPump_PlantLoop_EIR_CoolingFields::CondenserType, true);
+      OS_ASSERT(value);
+      return value.get();
+    }
+
+    bool HeatPumpPlantLoopEIRCooling_Impl::setCondenserType(const std::string& condenserType) {
+      bool ok = false;
+      if (openstudio::istringEqual("AirSource", condenserType) && (this->secondaryPlantLoop())) {
+        LOG(Warn, "Cannot set condenserType to AirSource, HeatPumpPlantLoopEIRCooling '"
+                    << this->name() << "' is connected to a Source Side Loop. Use removeFromSecondaryPlantLoop() instead.");
+      } else if (istringEqual("WaterSource", condenserType) && !(this->secondaryPlantLoop())) {
+        LOG(Warn, "Cannot set condenserType to 'WaterSource', HeatPumpPlantLoopEIRCooling '"
+                    << this->name() << "' is not connected to a Source Side Loop. Use addToNode(PlantLoop&) instead.");
+      } else {
+        ok = setString(OS_HeatPump_PlantLoop_EIR_CoolingFields::CondenserType, condenserType);
+      }
+
+      return ok;
+    }
+
+    boost::optional<HeatPumpPlantLoopEIRHeating> HeatPumpPlantLoopEIRCooling_Impl::companionHeatingHeatPump() const {
+      return getObject<ModelObject>().getModelObjectTarget<HeatPumpPlantLoopEIRHeating>(
+        OS_HeatPump_PlantLoop_EIR_CoolingFields::CompanionHeatPumpName);
+    }
+
+    boost::optional<double> HeatPumpPlantLoopEIRCooling_Impl::referenceLoadSideFlowRate() const {
+      return getDouble(OS_HeatPump_PlantLoop_EIR_CoolingFields::LoadSideReferenceFlowRate, true);
+    }
+
+    bool HeatPumpPlantLoopEIRCooling_Impl::isReferenceLoadSideFlowRateAutosized() const {
+      bool result = false;
+      boost::optional<std::string> value = getString(OS_HeatPump_PlantLoop_EIR_CoolingFields::LoadSideReferenceFlowRate, true);
+      if (value) {
+        result = openstudio::istringEqual(value.get(), "Autosize");
+      }
+      return result;
+    }
+
+    boost::optional<double> HeatPumpPlantLoopEIRCooling_Impl::referenceSourceSideFlowRate() const {
+      return getDouble(OS_HeatPump_PlantLoop_EIR_CoolingFields::SourceSideReferenceFlowRate, true);
+    }
```

Comment:
@joseph-robertson Why did you change the name of all these methods? That should be `sourceSideReferenceFlowRate` to match the IDD

Reply:
Did you manually copy from https://github.com/NREL/OpenStudio/blob/dd40326c4deeb34e3af2641535ffa4f65149db2f/src/model/HeatPumpWaterToWaterEquationFitCooling.cpp#L169-L172 ?

Reply:
Addressed in 3d1f5976b561c2f119d956313d9e333413beda74. It can be reverted if you feel extremely strongly about it.

Reply:
To match the convention of other objects. Why does the name of a method break something?